### PR TITLE
Sheets improves

### PR DESCRIPTION
**Pull request by denar90**
I added some options which can be passed from config file, so now we can ignore 5-th step from instruction list https://github.com/paulirish/pwmetrics/pull/18#issue-198892447

Next think is how to get rid of 
```
3. run node lib/gsheets/gsheets.js and follow instructions for authenicating
4. make a copy of this spreadsheet, remove rows 2-N.
```
Do you think this one https://github.com/google/google-api-nodejs-client/ can be helpful for case 3?

### CODE REVIEW

File: bin/cli.js [hunk 1]
```diff
@@ -17,15 +18,19 @@ argv
...
 if (flags.expectations) {
-  const expectationsConfig = getConfig(flags.expectations);
+  const expectationsConfig = getExpectationsConfig(flags.expectations);
   url = expectationsConfig.url;
   flags.metrics = expectationsConfig.metrics;
-} else {
-  url = argv.filter(f => !f.startsWith('-')).shift();
+} else if (flags.submit) {
+  flags.sheet = getSheetsConfig(flags.submit);
+  delete flags.submit;
```
Comment (denar90) [inline bin/cli.js]: I'm not sure about this one. Maybe it should be `submit` object which contains 
```
sheets: {
  type: ...,
  options: {}
}
```

Comment (denar90) [inline bin/cli.js]: Another think. I'll just leave it here. It's about flags `submit`, `expectations`. Right now for both of them users have to pass file path as value. What if we can do it better, like
`pwmetrics --submit --config=./path-to-config`
`pwmetrics --expecations --config=./path-to-config`
if path is not passed then config will be read from `package.json`.
Format of config file might be
```js
{
  url: 'http://example.com/',
  expectations: {
    metrics: {
      ttfcp: {
        warn: '>=30',
        error: '>=5000'
      },
      ttfmp: {
        warn: '>=3000',
        error: '>=50'
      },
      psi: {
        warn: '>=1500',
        error: '>=32'
      }
    }
  },
  sheets: {
    type: 'GOOGLE_SHEETS',
    options: {
      spreadsheetId: '1Uo4a4KQTaC1FBDPMIJ6xLMmOd2yLyps-6jBC5x4x6I8',
      tableName: 'data'
    }
  }
}

```

File: lib/gsheets/gsheets.js [hunk 2]
```diff
@@ -23,7 +21,7 @@ var TOKEN_PATH = TOKEN_DIR + 'sheets.googleapis.com-nodejs-pwmetrics.json';
 function authenticate() {
   return new Promise((resolve, reject) => {
     // Load client secrets from a local file.
-    fs.readFile('client_secret.json', 'utf8', function processClientSecrets(err, content) {
+    fs.readFile('./lib/gsheets/client_secret.json', 'utf8', function processClientSecrets(err, content) {
```
Comment (denar90) [inline lib/gsheets/gsheets.js]: Nice idea. Do you suggest to have this property value as path? Like
```
{
 ...
  clientSecretDataPath: '/my-global-path/client_secret.json'
 ...
}

Comment (paulirish) [inline lib/gsheets/gsheets.js]: eh. i'm thinking of like a `client_secret_contents` property and then just paste in the content of that original file.. so you just have one file total.

Comment (denar90) [inline lib/gsheets/gsheets.js]: Make sense. Thanks 👍

**Comment by paulirish** [comment]
merging this into the branch..

**Pull request merged**